analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/HashesShouldHaveUnpredictableSaltBase.cs

```diff
@@ -91,6 +104,23 @@ private static ProgramState ProcessArrayCreation(ProgramState state, IArrayCreat
     }
 
     private static bool IsCryptographicallyStrongRandomNumberGenerator(IInvocationOperationWrapper invocation) =>
-      (invocation.TargetMethod.Name.Equals(nameof(RandomNumberGenerator.GetBytes)) || invocation.TargetMethod.Name.Equals(nameof(RandomNumberGenerator.GetNonZeroBytes)))
-      && invocation.TargetMethod.ContainingType.DerivesFrom(KnownType.System_Security_Cryptography_RandomNumberGenerator);
+        (invocation.TargetMethod.Name.Equals(nameof(RandomNumberGenerator.GetBytes)) || invocation.TargetMethod.Name.Equals(nameof(RandomNumberGenerator.GetNonZeroBytes)))
+        && invocation.TargetMethod.ContainingType.DerivesFrom(KnownType.System_Security_Cryptography_RandomNumberGenerator);
+
+    private static IOperation FirstMethodArgument(ProgramState state, IInvocationOperationWrapper invocation, KnownType argumentType) =>
+        invocation.Arguments.FirstOrDefault(x => IsArgumentWithNameAndType(state, x, argumentType))?.AsArgument() is { } argument
+            ? state.ResolveCaptureAndUnwrapConversion(argument.Value)
+            : null;
+
+    private static IOperation FirstConstructorArgumentWithName(ProgramState state, IObjectCreationOperationWrapper objectCreation, KnownType argumentType, params string[] nameCandidates) =>
```

Comment:
I find params a bit dangerous and not specific. Better just add an array of string.
```suggestion
    private static IOperation FirstConstructorArgumentWithName(ProgramState state, IObjectCreationOperationWrapper objectCreation, KnownType argumentType, string[] nameCandidates) =>
```